CMakeLists.txt

```
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.21)
```

```suggestion
cmake_minimum_required(VERSION 3.22)
```
3.22 is the default in the Ubuntu 22.04 LTS and it is broadly available.
It is also the minimum required by Kokkos.

I would not go older than that.  I would be fine being more aggressive and requiring 3.28 which comes with the next long term support version of Ubuntu, that is 24.04